operators/pkg/instctrl/containers.go

```diff
@@ -28,6 +31,53 @@ import (
 	"github.com/netgroup-polito/CrownLabs/operators/pkg/utils"
 )
 
+const (
+	UserPvcSecret              = "user-pvc-secret"
+	UserPvcSecretServerIPKey   = "server-ip"
+	UserPvcSecretServerNameKey = "server-name"
+	UserPvcSecretPathKey       = "path"
+	NfsMountPath               = "/media/MyDrive"
+	NfsVolumeName              = "nfs-vol"
+)
+
+func (r *InstanceReconciler) GetNFSVolume(ctx context.Context, volume *v1.Volume, volumeMount *v1.VolumeMount) error {
+	// Retrieve the correct Tenant from the given context
+	instance := clctx.InstanceFrom(ctx)
+	if instance == nil {
+		err := errors.New("unable to retrieve instance from the context")
+		klog.Errorf("%s", err)
+		return err
+	}
+	// Get the secret and the NFS path
+	secret := v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: UserPvcSecret, Namespace: instance.Namespace}}
```

Comment:
The following should work as well, since you're providing information through NamespacedName. Defining the ObjectMeta is necessary only for the CreateOrUpdate.
```suggestion
	secret := v1.Secret{}
```

Reply:
Ok, removed